katapult_test.go

```diff
@@ -320,3 +363,126 @@ func Test_apiClient_Do(t *testing.T) {
 		})
 	}
 }
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		opts          []Opt
+		wantAPIKey    string
+		wantUserAgent string
+		wantBaseURL   *url.URL
+		wantTimeout   time.Duration
+		wantErr       string
+	}{
+		{
+			name:          "defaults",
+			wantAPIKey:    "",
+			wantUserAgent: DefaultUserAgent,
+			wantBaseURL:   DefaultURL,
+			wantTimeout:   DefaultTimeout,
+		},
+		{
+			name: "options specified",
```

Comment:
Can I suggest we have this test case specify all available options for the sake of extra paranoia? lol

Reply:
Hmm, I'm not sure I agree here. Each option already has its own test that checks it modifies the Client properly, and I think that we actually gain more by not specifying all the options as it ensures that the supplied options and the default options both come through properly, rather than it being all of one or all of the other.